### PR TITLE
feat(core/mdn-annotation): allow providing custom key

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes https://github.com/w3c/respec/issues/2860

### CODE REVIEW

File: tests/spec/core/mdn-annotation-spec.js [hunk 1]
```diff
@@ -2,17 +2,46 @@ import { flushIframes, makeRSDoc, makeStandardOps } from "../SpecHelper.js";
...
+    it("prefers `mdn.key` over shortName", async () => {
+      const ops = makeStandardOps({
+        mdn: { baseJsonPath, key: "payment-request" },
+        shortName: "whatever",
+      });
+      const doc = await makeRSDoc(ops, "spec/core/mdn-annotation.html");
+      expect(getMdnBox(doc).matches("aside.mdn")).toBeTrue();
+    });
+
+    it("supports mdn to be a string", async () => {
```
Comment (sidvishnoi) [inline tests/spec/core/mdn-annotation-spec.js:35]: This test sadly fetches over network, as we cannot set `mdn.baseJsonPath` (used for mock JSON source) on a string.
I will add a util for to know if we're in test env for these things in a separate PR.

Comment (sidvishnoi) [reply tests/spec/core/mdn-annotation-spec.js:35]: Well, all these tests are going over network right now anyway as tests don't add `mdn.specMapUrl` :man_facepalming:

Comment (marcoscaceres) [reply tests/spec/core/mdn-annotation-spec.js:35]: We can live with that...

**Comment by sidvishnoi** [comment]
Wiki updated: https://github.com/w3c/respec/wiki/mdn

**Pull request merged**